pkg/collector/corechecks/snmp/common/utils.go

```diff
@@ -42,3 +43,31 @@ func CopyStrings(tags []string) []string {
 func GetAgentVersionTag() string {
 	return "agent_version:" + version.AgentVersion
 }
+
+// NormalizeHost applies a liberal policy on host names.
+func NormalizeHost(host string) string {
```

Comment:
I think we should move it to https://github.com/DataDog/datadog-agent/blob/main/pkg/util/hostname.go (need Agent Core review) that contain hostname related util code.

`NormalizeHost` will be likely needed to normalise namespace/device_id from Traps and NetFlow.

Reply:
Moving to `hostname` has a larger impact, can potentially break current users. I'd rather start by adding it here, for snmp usage. Later have discussion with agent core team, we probably need to track down any user break

Reply:
> Moving to hostname has a larger impact, can potentially break current users. 

I meant just moving `NormalizeHost` to hostname.go, but we will be the only users of this method for now.

An alternative is to move `NormalizeHost` to https://github.com/DataDog/datadog-agent/tree/main/pkg/snmp where we have snmp related code (currently: snmp listener and traps).